requests/sessions.py

```diff
@@ -289,7 +289,9 @@ def rebuild_proxies(self, prepared_request, proxies):
         new_proxies = proxies.copy()
         no_proxy = proxies.get('no_proxy')
 
-        bypass_proxy = should_bypass_proxies(url, no_proxy=no_proxy)
+        bypass_proxy = False
+        if self.trust_env:
+            bypass_proxy = should_bypass_proxies(url, no_proxy=no_proxy)
         if self.trust_env and not bypass_proxy:
```

Comment:
Looking at the new change, I don't have an issue with the general idea. This seems like a general performance improvement for rebuild_proxies, regardless of the outcome for #5888. Stylistically, I'd like to simplify the logic here a bit more though. Something along the lines of this would be preferred:

```suggestion
        if self.trust_env and not should_bypass_proxies(url, no_proxy=no_proxy):
```

@omermizr I don't want to volunteer you but if you have a moment, would you mind confirming this PR resolves your issue in #5888?

Reply:
@nateprewitt While this is probably a good change, unless I'm missing something it doesn't resolve the issue - when 'trust_env' is set to True (which is the default), this changes nothing. 
More specifically to my use case - I have no control over how the third party package I'm using creates the session, and it uses the default value for 'trust_env' (like most usages).